src/combobox/index.ts

```diff
@@ -0,0 +1,721 @@
+
+let exportCombobox;
+(() => {
+  exportCombobox = window.customElements.get("kuc-combobox");
+  if (exportCombobox) {
+    return;
+  }
+
+  class KucCombobox extends KucBase {
+    @property({ type: String, reflect: true, attribute: "class" }) className =
+      "";
+    @property({ type: String }) error = "";
+    @property({ type: String, reflect: true, attribute: "id" }) id = "";
+    @property({ type: String }) label = "";
+    @property({ type: String }) value = "";
+    @property({ type: Boolean }) disabled = false;
+    @property({ type: Boolean }) requiredIcon = false;
+    @property({
+      type: Boolean,
+      attribute: "hidden",
+      reflect: true,
+      converter: visiblePropConverter,
+    })
+    visible = true;
+    @property({ type: Array }) items: ComboboxItem[] = [];
+
+    @state()
+    private _selectorVisible = false;
+
+    @query(".kuc-combobox__group")
+    private _groupEl!: HTMLDivElement;
+
+    @query(".kuc-combobox__group__toggle")
+    private _toggleEl!: HTMLDivElement;
+
+    @query(".kuc-combobox__group__toggle__input")
+    private _inputEl!: HTMLInputElement;
+
+    @query(".kuc-combobox__group__select-menu")
+    private _menuEl!: HTMLUListElement;
+
+    @queryAll(".kuc-combobox__group__select-menu__item")
+    private _itemsEl!: HTMLLIElement[];
+
+    @query(".kuc-combobox__group__label")
+    private _labelEl!: HTMLDivElement;
+
+    @query(".kuc-combobox__group__select-menu__item")
+    private _firstItemEl!: HTMLLIElement;
+
+    @query(".kuc-combobox__group__select-menu__item:last-child")
+    private _lastItemEl!: HTMLLIElement;
+
+    @query(".kuc-combobox__group__select-menu__item[aria-selected=true]")
+    private _selectedItemEl!: HTMLLIElement;
+
+    @query(".kuc-combobox__group__select-menu__highlight")
+    private _highlightItemEl!: HTMLLIElement;
+
+    @query(".kuc-base-error__error")
+    private _errorEl!: HTMLDivElement;
+
+    private _timeoutID!: number | null;
+
+    private _GUID: string;
+
+    @state()
+    private _searchText = "";
+
+    private _query = "";
+    private _matchingItems: ComboboxItem[] = [];
+
+    constructor(props?: ComboboxProps) {
+      super();
+      this._GUID = generateGUID();
+      const validProps = validateProps(props);
+      this._handleClickDocument = this._handleClickDocument.bind(this);
+      Object.assign(this, validProps);
+    }
+
+    shouldUpdate(changedProperties: PropertyValues): boolean {
+      if (changedProperties.has("items")) {
+        if (!validateItems(this.items)) {
+          throwErrorAfterUpdateComplete(this, ERROR_MESSAGE.ITEMS.IS_NOT_ARRAY);
+          return false;
+        }
+
+        const itemsValues = this.items.map((item) => item.value);
+        if (!validateDuplicatedValues(itemsValues)) {
+          throwErrorAfterUpdateComplete(
+            this,
+            ERROR_MESSAGE.ITEMS.IS_DUPLICATED
+          );
+          return false;
+        }
+      }
+
+      if (changedProperties.has("value")) {
```

Comment:
if (changedProperties.has("value") && !validateValueString(this.value)) {
...
}